client/dashboard/domains/domain-forwarding/form.tsx

```diff
@@ -176,8 +176,12 @@ export default function DomainForwardingForm( {
 				isValid: {
 					required: true,
 					custom: ( item ) => {
-						if ( ! isTargetUrlValid( item.targetUrl, domainName ) ) {
-							return __( 'Please enter a valid URL.' );
+						const validationError = isTargetUrlValid(
+							item.targetUrl,
+							item.subdomain ? `${ item.subdomain }.${ domainName }` : domainName
+						);
```

Comment:
When we swtich the form from `Subdomain` to `Domain`, the subdomain value in form state keeps the previously entered label, so the source domain is treated as subdomain.domain.com instead of the actual root:

https://github.com/user-attachments/assets/675c2b36-085d-42a5-a804-a1557d11e437

Here's a possible fix:

```suggestion
						const trimmedSubdomain = item.subdomain.trim();
						const sourceDomain =
							item.sourceType === '' && trimmedSubdomain
								? `${ trimmedSubdomain }.${ domainName }`
								: domainName;
						const validationError = isTargetUrlValid( item.targetUrl, sourceDomain );
```